app/models/user.rb

```diff
@@ -153,7 +153,7 @@ def active_mhv_ids
   end
 
   def address
-    address = mpi_profile&.address
+    address = identity&.address || mpi_profile&.address
```

Comment:
It looks like you'll have to change the structure of the `identity.address` object for this to work. Since it's a hash, you can't to `address.street`, etc, you'd need to change it to `address[:street]`, or, alternatively make it an `OpenStruct` in the definition in ssoe.rb